gnocchi/rest/aggregates/processor.py

```diff
@@ -167,16 +169,34 @@ def aggregated(timeseries, aggregation, from_timestamp=None,
                     'timespan (%.2f%%)' % (needed_percent_of_overlap,
                                            percent_of_overlap))
 
-        if aggregation in AGG_MAP:
-            values = AGG_MAP[aggregation](values, axis=1)
-        elif aggregation == 'count':
-            values = numpy.count_nonzero(~numpy.isnan(values), axis=1)
+        try:
+            granularity, times, values, is_aggregated = (
+                agg_operations.evaluate(operations, key, times, values,
+                                        False, references[key]))
+        except agg_operations.InvalidOperations as e:
+            raise UnAggregableTimeseries("Invalid operations: %s" % e.reason)
+
+        # NOTE(sileht): Check if an operations have aggregated the result.
+        # (aggregate, or binary operation between metrics, ...)
+        # Otherwise we tag each resulting serie with its original reference.
+        if len(values.shape) != 2:
```

Comment:
should the following code not be part of operations.evaluate()? seems weird we are verify results in a different part of code.

Reply:
I have moved the verify where the issue can occurs.